src/vip/batch_address_test_tool/processor.clj

```diff
@@ -70,11 +71,23 @@
 
 (def retrieve-polling-locations (->pass-through retrieve-polling-locations*))
 
-(defn calculate-match*
+(defn calculate-scores*
   [ctx]
-  ctx)
+  (let [addresses (:addresses ctx)
+        scores (map match/calculate-score addresses)
+        merged (map #(assoc %1 :score %2) addresses scores)]
+    (assoc ctx :addresses merged)))
+
+(def calculate-scores (->pass-through calculate-scores*))
+
+(defn calculate-matches*
+  [ctx]
+  (let [addresses (:addresses ctx)
+        matches (map match/calculate-match addresses)
+        merged (map #(assoc %1 :match %2) addresses matches)]
+    (assoc ctx :addresses merged)))
 
-(def calculate-match (->pass-through calculate-match*))
+(def calculate-matches (->pass-through calculate-matches*))
```

Comment:
`calculate-scores*` and `calculate-matches*` are identical in form, with just a couple of details changed. This could be a good opportunity for Functional Programming:tm:.

In a final state, I might have done something like this:

```clj
;; in vip.batch-address-test-tool.match
(defn assoc-result-of-f
  "Given a key and function, returns a function taking a map which
  assocs the result of f on the map to the key.

  (def add-keys (assoc-result-of-f :ks keys))
  (add-keys {:a 1 :b 2})
  ;;=> {:a 1 :b 2 :ks [:a :b]}"
  [k f]
  (fn [m]
    (assoc m k (f m))))

(def add-score (assoc-result-of-f :score calculate-score))
(def add-match (assoc-result-of-f :match calculate-match))
```

```clj
;; in vip.batch-address-test-tool.processor
(defn map-key-in-ctx
  "Given a key and a function, returns a function that takes a context
  and updates the value at the key with the result of mapping the
  function over that value.

  (def stringify-lucky-numbers (map-key-in-ctx :lucky-numbers str))
  (stringify-lucky-numbers {:a 1 :lucky-numbers [8 13]}
  ;;=> {:a :lucky-numbers [\"8\" \"13\"}"
  [k f]
  (fn [ctx]
    (update ctx k (partial map f))))

(def calculate-scores*
  (map-key-in-ctx :addresses match/add-score))

(def calculate-matches*
  (map-key-in-ctx :addresses match/add-match))
```

Now, this _might be_ overkill. So maybe just take it as a "here's another way this could have been done". Or perhaps a halfway point between what you have and what I've presented looks good to you. Happy to talk about it.